lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+class OmiseCapabilities extends OmiseApiResource
+{
+    const ENDPOINT = 'capability';
+
+    /**
+     * Retrieves capabilities.
+     *
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @return OmiseCapabilities
+     */
+    public static function retrieve($publickey = null, $secretkey = null)
+    {
+        return parent::g_retrieve(get_class(), self::getUrl(), $publickey, $secretkey);
+    }
+
+    /**
+     * Retrieves array of payment backends. Optionally pass in as many filter functions as you want
+     *
+     * @param [function,...]
+     *    
+     * @return array
+     */
+    public function getBackends() {
+        // check for filters
+        if ($filters = func_get_args()) $filter = self::combineFilters($filters);
+        $res = $this['payment_backends'];
+        array_walk(
+            $res,
+            function($v, $k) use (&$res) {
+                $id = array_keys($v)[0];
+                $res[$k][$id]['_id'] = $id;
+            }
+        );
+        $res = array_map(function($a) { return (object)reset($a);}, $res);
+        return !empty($filter) ? array_filter($res, $filter) : $res;
+    }
+
+    /**
+     * Makes a filter function to check supported currency for backend.
+     *
+     * @param  string $currency
+     *
+     * @return function
+     */
+    public function backendSupportsCurrency($currency) {
+        return function($backend) use ($currency) { return in_array($currency, $backend->currencies); };
+    }
+
+    /**
+     * Makes a filter function to check type of backend.
+     *
+     * @param  string $type
+     *
+     * @return function
+     */
+    public function backendTypeIs($type) {
+        return function($backend) use ($type) { return $backend->type==$type; };
+    }
+
+    /**
+     * Makes a filter function to check if backends can handle given amount.
+     *
+     * @param  int $amount
+     *
+     * @return function
+     */
+    public function backendSupportsChargeAmount($amount) {
```

Comment:
@jonrandy I have a feeling that this will cause of confusion in case if someone using this filter with `card` backend-type.
```php
$capability->getBackends(
    $capability->backendTypeIs('card'),
    $capability->backendSupportsChargeAmount(3000)
);
```

What do you think about this case?

Reply:
Again, sorry seems I misunderstood.. let me test a bit more..

Reply:
Every backend has a min/max charge amount. So the filter makes sense for all

Reply:
also `3000` to `const` ?